clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -83,69 +82,14 @@ interface __AnalysisApi {
   getControlFlowAnalysis(documentUri: string): Promise<AnalysisResult>;
 }
 
-let languageClientService: LanguageClientService;
-let controlFlowChannel: vscode.LogOutputChannel;
-let analysisService: ControlFlowAnalysisService;
 const API_VERSION: string = "1.0.1";
 
-async function initialize(context: vscode.ExtensionContext) {
-  // We need lazy initialization to be able to mock this for unit testing
-  controlFlowChannel = vscode.window.createOutputChannel(
-    "COBOL Language Support Control Flow",
-    { log: true },
-  );
-
-  analysisService = new ControlFlowAnalysisService(
-    outputChannel,
-    controlFlowChannel,
-  );
-  try {
-    await vscode.workspace.fs.createDirectory(context.globalStorageUri);
-  } catch (error) {
-    const message = `${FAIL_CREATE_GLOBAL_STORAGE_MSG}: ${getErrorMessage(
-      error,
-    )}`;
-    outputChannel.appendLine(message);
-    throw Error(message);
-  }
-
-  languageClientService = new LanguageClientService(
-    outputChannel,
-    context.globalStorageUri,
-    {
-      executeCommand: (command, args, next) => {
-        if (command == "missing copybook") {
-          externalApis.clearProfiles();
-        }
-        next(command, args);
-      },
-    },
-  );
-
-  await initializeExternalAPIs(
-    context.globalStorageUri,
-    languageClientService.invalidateConfiguration,
-  );
-
-  const configurationWatcher = new ConfigurationWatcher();
-  const dialectService = new DialectService(
-    languageClientService,
-    outputChannel,
-  );
-
-  return {
-    configurationWatcher,
-    dialectService,
-  };
-}
-
 export async function activate(
   context: vscode.ExtensionContext,
 ): Promise<__ExtensionApi & __AnalysisApi> {
+  await createExtensionFolder(context);
+
   await initTelemetry(context);
```

Comment:
```suggestion
  await initTelemetry(context);

  await createExtensionFolder(context);
```
so that errors get reported.

Reply:
That would make sense. I am not sure there is any capturing of exceptions by default though. Do you want me to add a `telemetryExceptionEvent` in case creation of the extension folder fails?